hubble/cmd/common/k8s/client.go

```diff
@@ -0,0 +1,142 @@
+package k8s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/httpstream"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport/spdy"
+
+	"github.com/cilium/cilium/pkg/safeio"
+)
+
+// Slim version of: https://github.com/cilium/cilium/tree/v1.17.0-pre.1/cilium-cli/k8s/client.go
+type Client struct {
+	Clientset kubernetes.Interface
+	Config    *rest.Config
+}
+
+func NewClient(contextName, kubeconfig string) (*Client, error) {
+	restClientGetter := genericclioptions.ConfigFlags{
+		Context:    &contextName,
+		KubeConfig: &kubeconfig,
+	}
+	rawKubeConfigLoader := restClientGetter.ToRawKubeConfigLoader()
+
+	config, err := rawKubeConfigLoader.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		Clientset: clientset,
+		Config:    config,
+	}, nil
+}
+
+func (c *Client) GetService(ctx context.Context, namespace, name string, opts metav1.GetOptions) (*corev1.Service, error) {
+	return c.Clientset.CoreV1().Services(namespace).Get(ctx, name, opts)
+}
+
+// TODO: Consider replacing kubectl-based port-forwarding with native impl using k8s client
+func (c *Client) ProxyTCP(ctx context.Context, namespace, name string, port uint16, handler func(io.ReadWriteCloser) error) error {
```

Comment:
This looks great, but if we're not going to use it in the initial PR, I suggest we just remove it for now and add it in a follow up PR.

Reply:
Makes sense, will do.